tests/assembly/asm/inline-asm-avx.rs

```diff
@@ -0,0 +1,25 @@
+// assembly-output: emit-asm
+// compile-flags: --crate-type=lib
+// compile-flags: --target x86_64-unknown-linux-gnu
+// needs-llvm-components: x86
+
+#![feature(portable_simd)]
+
+use std::simd::Simd;
+use std::arch::asm;
+
+#[target_feature(enable = "avx")]
+#[no_mangle]
+// CHECK-LABEL: convert:
+pub unsafe fn convert(a: *const f32) -> Simd<f32, 8> {
+    // CHECK: vbroadcastss (%rsi), {{%ymm[0-7]}}
```

Comment:
I don't believe the specific scalar register is important.
```suggestion
    // CHECK: vbroadcastss (%{{[er][a-ds0-9][xpi0-9]?}}), {{%ymm[0-7]}}
```